File: sdk/examples/python/chronovet.py
#!/usr/bin/env python3
"""E2E example: drive ghostchrome from the Python SDK against a live site.

Attaches to an already-running Chrome via --connect=auto (project policy).

    GHOSTCHROME_BIN="$PWD/ghostchrome" python3 sdk/examples/python/chronovet.py [url]
"""
import os
import sys

# Make the in-repo SDK importable without installing it.
sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", "..", "python"))

from ghostchrome import Ghostchrome  # noqa: E402

url = sys.argv[1] if len(sys.argv) > 1 else "https://www.chronovet.fr/"
binary = os.environ.get("GHOSTCHROME_BIN", "ghostchrome")
connect = os.environ.get("GHOSTCHROME_CONNECT", "--connect=auto")

with Ghostchrome(command=binary, extra_flags=[connect, "--stealth"]) as gc:
    nav, _ = gc.navigate(url)
    print(f"[{nav.status}] {nav.title}")
    print(f"url: {nav.url}")

    tree, _ = gc.extract(level="skeleton")
    refs = tree.refs or {}
    print(f"refs: {len(refs)}")
    for ref, node in list(refs.items())[:10]:
        role = node.get("role", "")
        name = node.get("name", "")
        print(f"  {ref} {role} {name}".rstrip())

File: sdk/examples/typescript/chronovet.ts
/**
 * E2E example: drive ghostchrome from the TypeScript SDK against a live site.
 *
 * Attaches to an already-running Chrome via --connect=auto (project policy).
 *
 *   GHOSTCHROME_BIN="$PWD/ghostchrome" bun run sdk/examples/typescript/chronovet.ts [url]
 */
import { createGhostchrome } from "../../typescript/src/index";

const url = process.argv[2] ?? "https://www.chronovet.fr/";
const bin = process.env.GHOSTCHROME_BIN ?? "ghostchrome";
const connect = process.env.GHOSTCHROME_CONNECT ?? "--connect=auto";

const gc = createGhostchrome({ command: bin, flags: [connect, "--stealth"] });

try {
  const nav = await gc.navigate(url);
  console.log(`[${nav.result.status}] ${nav.result.title}`);
  console.log(`url: ${nav.result.url}`);

  const ex = await gc.extract({ level: "skeleton" });
  const refs = ex.result.refs ?? {};
  const entries = Object.entries(refs);
  console.log(`refs: ${entries.length}`);
  for (const [ref, node] of entries.slice(0, 10)) {
    console.log(`  ${ref} ${node.role ?? ""} ${node.name ?? ""}`.trimEnd());
  }
} finally {
  await gc.close();
}
